oauth2-core/src/main/java/org/springframework/security/oauth2/core/protocol/ResponseAttributesExtractor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.protocol;
+
+import org.springframework.security.oauth2.core.OAuth2Attributes;
+import org.springframework.util.Assert;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Joe Grandja
+ */
+public final class ResponseAttributesExtractor {
```

Comment:
I don't think this should be static classes. Instead perhaps consider implementing the Converter (or similar) API.